:memo: DOC: Improve Installation Instructions (#567)
TissueImageAnalytics/tiatoolbox: docs/installation.rst

************

Prerequisites
=============
The prerequisites for tiatoolbox installation are OpenSlide binaries and OpenJpeg version 2.3.0 or above.
Please follow the instructions below to install prerequisite software according to the platform you are using.

Linux (Ubuntu)
--------------
On Linux the prerequisite software can be installed using the command

.. code-block:: console

    $ apt-get -y install libopenjp2-7-dev libopenjp2-tools openslide-tools

The same command is used when working on the Colab or Kaggle platforms.
When working on Google Colab, we remove the packages ``datascience`` and ``albumentations`` because they conflict
and produce an error message.

> Windows (10 and 11)

Author: ```suggestion
Windows (10+)
```

    C:\> conda install -c conda-forge openjpeg

macOS
-----

On macOS there are two popular package managers, `homebrew`_ and `macports`_.

.. _homebrew: https://brew.sh/
.. _macports: https://www.macports.org/

Homebrew
^^^^^^^^

.. code-block:: console

    $ brew install openjpeg openslide

MacPorts
^^^^^^^^

.. code-block:: console

    $ port install openjpeg openslide

> Installing Stable release (Recommended)

Reviewer: ```suggestion
Installing Stable Release (Recommended)
```

=======================================

Please note that TIAToolbox is tested for python version 3.8, 3.9 and 3.10.

> To install TIA Toolbox, run this command in your terminal after you have installed the prerequisite software:

Reviewer: ```suggestion
To install TIAToolbox, run this command in your terminal after you have installed the prerequisite software:
```

    C:\> conda install -c conda-forge openjpeg

macOS
-----

On macOS there are two popular package managers, `homebrew`_ and `macports`_.

.. _homebrew: https://brew.sh/
.. _macports: https://www.macports.org/

Homebrew
^^^^^^^^

.. code-block:: console

    $ brew install openjpeg openslide

MacPorts
^^^^^^^^

.. code-block:: console

    $ port install openjpeg openslide

> Installing Stable Release (Recommended)

Author: ```suggestion
Installing Stable Release
```

> =======================================

Author: ```suggestion
=========================
```

Please note that TIAToolbox is tested for python version 3.8, 3.9 and 3.10.

Recommended

> ---

Author: ```suggestion
-----------
```

> New

Reviewer 2: I don't think having a subheader of "New" is required here. Just have subheaders for upgrades and without dependencies.

Author: ok let me change it to Recommended and remove recommended from the title.

Author: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/567/commits/d221593ae51a15d2743d3b60a0bcb2ceee7344fb

> New

Author: ```suggestion
Recommended
```

    $ pip install --ignore-installed --upgrade tiatoolbox

If you already have setup a Python environment with all the pre-requisite software and dependencies installed and you would like to keep the existing versions of these dependencies, run this command in your terminal:

.. code-block:: console

    $ pip install --no-deps tiatoolbox

If you don't have `pip`_ installed, this `Python installation guide`_ can guide
you through the process.

.. _pip: https://pip.pypa.io
.. _Python installation guide: http://docs.python-guide.org/en/latest/starting/installation/

Alternative Method(s)
=====================

Using Anaconda
--------------

> After `installing Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or miniconda), you can install TIA toolbox using the following command:

Reviewer: Add link to miniconda?

Author: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/567/commits/6f77826d6fc71f40b554e4c36f7e1c97db97f698

> After `installing Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or miniconda), you can install TIA toolbox using the following command:

Reviewer: ```suggestion
After `installing Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or miniconda / mamba), you can install TIA toolbox using the following command:
```

Reviewer: Mamba: https://mamba.readthedocs.io/en/latest/user_guide/mamba.html

Author: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/567/commits/187f81540e45c1faae08649dcc710a4090812eda

> After `installing Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or miniconda), you can install TIA toolbox using the following command:

Author: ```suggestion
After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`), you can install TIA toolbox using the following command:
```

> After `installing Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or miniconda), you can install TIA toolbox using the following command:

Author: Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/567/commits/6f77826d6fc71f40b554e4c36f7e1c97db97f698

> After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`), you can install TIA toolbox using the following command:

Author: ```suggestion
After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ ), you can install TIAToolbox using the following command:
```

> After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ ), you can install TIAToolbox using the following command:

Author: ```suggestion
After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ / `mamba <https://mamba.readthedocs.io/en/latest/user_guide/mamba.html>`_ ), you can install TIAToolbox using the following command:
```

> After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ / `mamba <https://mamba.readthedocs.io/en/latest/user_guide/mamba.html>`_ ), you can install TIAToolbox using the following command:

Author: ```suggestion
After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ / `mamba <https://mamba.readthedocs.io/en/latest/user_guide/mamba.html#mamba-vs-conda-clis>`_ ), you can install TIAToolbox using the following command:
```

> After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ / `mamba <https://mamba.readthedocs.io/en/latest/user_guide/mamba.html#mamba-vs-conda-clis>`_ ), you can install TIAToolbox using the following command:

Reviewer: ```suggestion
After installing `Anaconda <https://docs.anaconda.com/anaconda/install/index.html>`_ (or `miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ / `mamba <https://mamba.readthedocs.io/en/latest/user_guide/mamba.html#mamba-vs-conda-clis>`_), you can install TIAToolbox using the following command:
```

Just removing the space at the end of the brackets.

.. _tarball: https://github.com/TissueImageAnalytics/tiatoolbox/tarball/master

Using Docker
------------

To run TIA toolbox in an isolated environment, use our `Docker image <https://github.com/tissueimageanalytics/tiatoolbox-docker/pkgs/container/tiatoolbox>`_ . We host different Dockerfiles in our github repository `tiatoolbox-docker <https://github.com/TissueImageAnalytics/tiatoolbox-docker>`_. Please report any issues related to the docker image in the repository `tiatoolbox-docker <https://github.com/TissueImageAnalytics/tiatoolbox-docker>`_.

After `installing Docker <https://docs.docker.com/get-docker/>`_ (or Docker Desktop), you can use our TIA toolbox image in 3 different ways.

Use the Pre-Built Docker Image
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 1. Pull the Image From Github Container Registry

Author: TIAToolbox provides pre-built docker containers which can be downloaded using the instructions below:

> 1. Pull the Image From Github Container Registry

Author: ```suggestion
TIAToolbox provides pre-built docker containers which can be downloaded using the instructions below:
1. Pull the Image From Github Container Registry
```

.. _tarball: https://github.com/TissueImageAnalytics/tiatoolbox/tarball/master

Using Docker
------------

To run TIA toolbox in an isolated environment, use our `Docker image <https://github.com/tissueimageanalytics/tiatoolbox-docker/pkgs/container/tiatoolbox>`_ . We host different Dockerfiles in our github repository `tiatoolbox-docker <https://github.com/TissueImageAnalytics/tiatoolbox-docker>`_. Please report any issues related to the docker image in the repository `tiatoolbox-docker <https://github.com/TissueImageAnalytics/tiatoolbox-docker>`_.

After `installing Docker <https://docs.docker.com/get-docker/>`_ (or Docker Desktop), you can use our TIA toolbox image in 3 different ways.

Use the Pre-Built Docker Image
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TIAToolbox provides pre-built docker containers which can be downloaded using the instructions below:

> 1. Pull the Image From Github Container Registry

Author: ```suggestion

1. Pull the Image From Github Container Registry
```

> Use the pre-built docker image

Author: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/567/commits/0489b845109b8e3702b11a1b27eac82bf6dce012

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 1. Pull the image from the Github Container Registry

Author: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/567/commits/0489b845109b8e3702b11a1b27eac82bf6dce012

""""""""""""""""""""""""""""""""""""""""""""""""""""
.. code-block:: console

    $ docker pull ghcr.io/tissueimageanalytics/tiatoolbox:latest

> 2. Use the pre-built Docker image as a base image in a Dockerfile

Author: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/567/commits/1ba477930f4a4bb0b30d55f24a08111a756fe4cb
